app/gui/src/project-view/util/codemirror/__tests__/markdownEditing.test.ts

```diff
@@ -0,0 +1,274 @@
+    source: '- Li|st item\n- List item\n- Lis|t item',
+    expected: 'List item\nList item\nList item',
+  },
+  {
+    desc: 'Change ordered list to unordered list',
+    source: '1. List| item\n2. List item\n3. Lis|t item',
+    expected: '- List item\n- List item\n- List item',
+  },
+  {
+    desc: 'Disable unordered list in code block',
+    source: '```\nSome code\n- Lis|t item\nMore code\n```',
+    expected: '```\nSome code\nList item\nMore code\n```',
+  },
+  {
+    desc: 'Create unordered list in code block',
+    source: '```\nSome code\nLis|t item\nAnother |list item\n```',
+    expected: '```\nSome code\n- List item\n- Another list item\n```',
+  },
+  {
+    desc: 'Change ordered list to unordered list in code block',
+    source: '```\nSome code\n1. List| item\n2. List item\n3. Lis|t item\nSome paragraph\n```',
+    expected: '```\nSome code\n- List item\n- List item\n- List item\nSome paragraph\n```',
+  },
+  {
+    desc: 'Disable unordered list in code block',
+    source: '```\nSome code\n- List| item\n- List item\n- Lis|t item\nSome paragraph\n```',
+    expected: '```\nSome code\nList item\nList item\nList item\nSome paragraph\n```',
+  },
+]
+
+test.each(unorderedListTestCases)('markdown unordered list $desc', ({ source, expected }) => {
+  const view = setupEditor(source)
+  toggleList(view, 'unordered')
+  expect(view.state.doc.toString()).toEqual(expected)
+})
+
+const orderedListTestCases: TestCase[] = [
+  {
+    desc: 'Create unordered list from empty line',
+    source: '|',
+    expected: '1. ',
+  },
+  {
+    desc: 'Create simple ordered list',
+    source: 'Li|st item\nList item\nLis|t item',
+    expected: '1. List item\n2. List item\n3. List item',
+  },
+  {
+    desc: 'Disable ordered list',
+    source: '1. Li|st item\n2. List item\n3. Lis|t item',
+    expected: 'List item\nList item\nList item',
+  },
+  {
+    desc: 'Change unordered list to ordered list',
+    source: '- List| item\n- List item\n- Lis|t item',
+    expected: '1. List item\n2. List item\n3. List item',
+  },
+  {
+    desc: 'Create ordered list in code block',
+    source: '```\nSome code\nLis|t item\nAnother |list item\n```',
+    expected: '```\nSome code\n1. List item\n2. Another list item\n```',
+  },
+  {
+    desc: 'Change unordered list to ordered list in code block',
+    source: '```\nSome code\n- List| item\n- List item\n- Lis|t item\nSome paragraph\n```',
+    expected: '```\nSome code\n1. List item\n2. List item\n3. List item\nSome paragraph\n```',
+  },
+  {
+    desc: 'Disable ordered list in code block',
+    source: '```\nSome code\n1. List| item\n2. List item\n3. Lis|t item\nSome paragraph\n```',
+    expected: '```\nSome code\nList item\nList item\nList item\nSome paragraph\n```',
+  },
+]
+
+test.each(orderedListTestCases)('markdown ordered list $desc', ({ source, expected }) => {
+  const view = setupEditor(source)
+  toggleList(view, 'ordered')
+  expect(view.state.doc.toString()).toEqual(expected)
+})
```

Comment:
I don't see any tests for checking what the "current" block type is identified as when different text is selected--that's complex and I think important to cover.

Reply:
As discussed, not needed. We might want to implement it in the future, but I don’t see much point given instant visual feedback in the editor, highlighting the block markers.